### DISCUSSION TITLE
TCmenu 3 sub menus

### DISCUSSION THREAD

**Discussion by tomelgato** [davetcc/tcMenu#318] [Q&A] [answered]
I have a menustructure with submenus. When i go into a submenu i dont see a cursor. I need to press down/next, then the cursor appears in the 2nd line (1st item of submenu). I need to go up again to go to the submenutitle to be able to go up in menustructure again (before i dont see the [..] too).
Similar behavior when reaching the end of the submenu and jumping to 1st item again

Long story short ... it seems that the 1st submenu item (the title = backfunction) is disabled when going into a submenu

**Comment by davetcc** [comment]
Sorry but there is insufficient detail here, please provide images showing the situation, which display you are targetting, what input technology, depth of menu structure etc

**Comment by tomelgato** [comment]
Sorry, youre right. Will do a video next time when im at the machine.

4x20 LCD with 4x5 matrix keyboard. It happens at the first sub level already. Its like the cursor is deactivated when you enter the submenu and it gets shown after the first valid keystroke.
Or maybe the first entry (Submenu Title) is not active when you enter because there is no [..] too

SUBMENU

when you go down and up again its complete and works 

SUBMENU    [..]

**Comment by davetcc** [comment]
Even a screen shot of the situation with a description would be helpful. 
Thanks

**Comment by davetcc** [comment]
Moved to a discussion for now while we work out what the issue is. I'll create an issue if needed.

**Reply by tomelgato** [reply]
> Moved to a discussion for now while we work out what the issue is. I'll create an issue if needed.

Hey dave, did upload the bug video

**Reply by davetcc** [reply]
Hi thanks for that, I'm going to look at this next, just been trying to get 3.2 BETA out first and some known bugs fixed.

**Reply by tomelgato** [reply]
Is this confirmed / solved in 4.0?

**Reply by davetcc** [reply]
Not at the moment, ran out of time in 4.0 to track it down, but the good news is that going forward the main priority is now bug fixing, API renovation, and remote capabilities. Meaning that this and other small issues are now the priority.

**Reply by tomelgato** [reply]
Maybe you move this out of discussion to not forget it for nxt release ;)

**Comment by tomelgato** [comment]
Sorry, did take a while. Heres a small video


https://user-images.githubusercontent.com/92522434/230735470-f404e5c4-356c-4390-b02d-f30f92bc4446.mp4

**Comment by davetcc** [comment, marked as answer]
I've made https://github.com/davetcc/tcMenu/issues/308 a catch all for a few rendering issues that need attention.
